lecture-notes/2-vcs.md

```diff
@@ -585,19 +585,79 @@ commit message style. Nonetheness, most projects agree that consistent styling
 across commits is a good thing, as it makes commit histories easier to read and
 makes collaboration easier.
 
-TODO: New example with the commands used last lecture, but leave out the
-`.git/` stuff and instead show multiple iterations of `git add` and `git
-commit` with realistic changes and commit messages.
-
-#### Staging files partially (`git add -p`)
-
-TODO: Talk about how, in practice, you often make multiple changes at a time
-before thinking about committing. Adding at the file level (or worse, at the
-repo level with `git add .`) almost always includes more than you want, plus it
-doesn't let you double-check your changes.
+### Staging files partially (`git add -p`)
+If you try to follow our advice using the commands we've covered so far, you'll
+soon find that it's very difficult to create tidy commits when you can only
+stage changes on a file-by-file basis (`git add <file>`). The process of
+software development is rarely as organized as a Git history should be: while
+working on one change, for example, you might see other issues in the same file
+and fix them on the fly.
+
```

Comment:
suggestion: Kafka  Metamorphosis where you fix typos and also go from raven to bug because raven is taken already